File: ViewModel/Screens/TeamDialogViewModel.vb
Imports POFF.Kicker.ViewModel.Types

Namespace Screens

    Public Class TeamDialogViewModel

        Public Sub New(teamInfo As TeamInfo)
            ' Check parameters
            If teamInfo Is Nothing Then Throw New ArgumentNullException("teamInfo")

            ' Initialize members
            TeamInfoValue = teamInfo
        End Sub

        Private ReadOnly TeamInfoValue As TeamInfo
        Private ReadOnly Property TeamInfo As TeamInfo
            Get
                Return TeamInfoValue
            End Get
        End Property

        Public Sub AcceptChanges()
            TeamInfo.AcceptChanges()
        End Sub

    End Class

End Namespace

File: View/Screens/TeamDialogViewModel.cs
using System;
using POFF.Kicker.ViewModel.Types;

namespace POFF.Kicker.ViewModel.Screens
{

    public class TeamDialogViewModel
    {

        public TeamDialogViewModel(TeamInfo teamInfo)
        {
            // Check parameters
            if (teamInfo is null)
                throw new ArgumentNullException("teamInfo");

            // Initialize members
            TeamInfoValue = teamInfo;
        }

        private readonly TeamInfo TeamInfoValue;
        private TeamInfo TeamInfo
        {
            get
            {
                return TeamInfoValue;
            }
        }

        public void AcceptChanges()
        {
            TeamInfo.AcceptChanges();
        }

    }

}